hs_geo_raster_resource/models.py

```diff
@@ -108,28 +108,29 @@ def update(cls, element_id, **kwargs):
     def remove(cls, element_id):
         raise ValidationError("Coverage element can't be deleted.")
 
-    def add_to_xml_container(self, container):
-        """Generates xml+rdf representation of the metadata element"""
-
-        NAMESPACES = CoreMetaData.NAMESPACES
-        cov = etree.SubElement(container, '{%s}spatialReference' % NAMESPACES['hsterms'])
-        cov_term = '{%s}' + 'box'
-        coverage_terms = etree.SubElement(cov, cov_term % NAMESPACES['hsterms'])
-        rdf_coverage_value = etree.SubElement(coverage_terms,
-                                              '{%s}value' % NAMESPACES['rdf'])
-        # raster original coverage is of box type
-        cov_value = 'northlimit=%s; eastlimit=%s; southlimit=%s; westlimit=%s; units=%s' \
-                    % (self.value['northlimit'], self.value['eastlimit'],
-                       self.value['southlimit'], self.value['westlimit'],
-                       self.value['units'])
-
-        for meta_element in self.value:
-            if meta_element == 'projection':
-                cov_value += '; projection_name={}'.format(self.value[meta_element])
-            if meta_element in ['projection_string', 'datum']:
-                cov_value += '; {}={}'.format(meta_element, self.value[meta_element])
-
-        rdf_coverage_value.text = cov_value
+    hsterms = ['spatialReference', 'box', ]
+    rdf = ['value']
+
+    def rdf_triples(self, subject, graph):
+        original_coverage = BNode()
+        graph.add((subject, HSTERMS.spatialReference, original_coverage))
+        value = BNode()
+        graph.add((original_coverage, HSTERMS.box, value))
+        value_string = "; ".join(["=".join([key, str(val)]) for key, val in self.value.items()])
+        graph.add((value, RDF.value, Literal(value_string)))
+
+    @classmethod
+    def ingest_rdf(cls, graph, subject, content_object):
+        spatial_object = graph.value(subject=subject, predicate=HSTERMS.spatialReference)
+        box_object = graph.value(subject=spatial_object, predicate=HSTERMS.box)
+        value_str = graph.value(subject=box_object, predicate=RDF.value)
+        if value_str:
+            value_str = value_str.value
+            value_dict = {}
+            for p in value_str.split('; '):
+                k, v = p.split('=')
```

Comment:
It might be better to split by ```;``` without the space and then strip p. This will be less sensitive to potential future change, e.g., no space after the colon.

Reply:
that's a good idea.

Reply:
I updated the resource and feature metadata, somehow missed raster until now.